diff -ru pycurl-7.19.5-orig/setup.py pycurl-7.19.5/setup.py
--- pycurl-7.19.5-orig/setup.py	Sat Jul 12 22:47:30 2014
+++ pycurl-7.19.5/setup.py	Wed Aug 13 16:43:21 2014
@@ -147,7 +147,8 @@
         optbuf = ''
         sslhintbuf = ''
         errtext = ''
-        for option in ["--libs", "--static-libs"]:
+	# NOTE - OmniOS build doesn't want --static-libs AT ALL!!!
+        for option in ["--libs"]:
             p = subprocess.Popen((CURL_CONFIG, option),
                 stdout=subprocess.PIPE, stderr=subprocess.PIPE)
             stdout, stderr = p.communicate()
