--- jsonrpclib-0.1.3/jsonrpclib//jsonclass.py   Tue Oct 26 01:38:47 2010
+++ jsonrpclib/jsonrpclib//jsonclass.py Wed Nov 19 17:26:16 2014
@@ -129,6 +129,13 @@
         except ImportError:
             raise TranslationError('Could not import %s from module %s.' %
                                    (json_class_name, json_module_tree))
+
+        # The returned class is the top-level module, not the one we really
+        # want.  (E.g., if we import a.b.c, we now have a.)  Walk through other
+        # path components to get to b and c.
+        for i in json_module_parts[1:]:
+            temp_module = getattr(temp_module, i)
+
         json_class = getattr(temp_module, json_class_name)
     # Creating the object...
     new_obj = None
