--- ./builds/unix/freetype-config.in.orig	2012-03-06 22:02:53.372084321 +0000
+++ ./builds/unix/freetype-config.in	2012-03-06 22:03:47.053523682 +0000
@@ -131,7 +131,7 @@
 fi
 
 if test "$echo_cflags" = "yes" ; then
-  cflags="-I${SYSROOT}$includedir/freetype2"
+  cflags="-I${SYSROOT}$includedir/freetype2 -I${SYSROOT}$libdir/freetype-2.4.8/include"
   if test "${SYSROOT}$includedir" != "/usr/include" ; then
     echo $cflags -I${SYSROOT}$includedir
   else
