--- builds/unix/install.mk.orig	2012-03-06 22:07:32.076863046 +0000
+++ builds/unix/install.mk	2012-03-06 22:07:08.278749340 +0000
@@ -30,6 +30,7 @@
 install: $(PROJECT_LIBRARY)
 	$(MKINSTALLDIRS) $(DESTDIR)$(libdir)                               \
                          $(DESTDIR)$(libdir)/pkgconfig                     \
+                         $(DESTDIR)$(libdir)/freetype-2.4.8/include/freetype/config \
                          $(DESTDIR)$(includedir)/freetype2/freetype/config \
                          $(DESTDIR)$(includedir)/freetype2/freetype/cache  \
                          $(DESTDIR)$(bindir)                               \
@@ -51,7 +52,7 @@
 	$(INSTALL_DATA) $(BUILD_DIR)/ft2unix.h \
           $(DESTDIR)$(includedir)/ft2build.h
 	$(INSTALL_DATA) $(OBJ_BUILD)/ftconfig.h                        \
-          $(DESTDIR)$(includedir)/freetype2/freetype/config/ftconfig.h
+          $(DESTDIR)$(libdir)/freetype-2.4.8/include/freetype/config/ftconfig.h
 	$(INSTALL_DATA) $(OBJ_DIR)/ftmodule.h                          \
           $(DESTDIR)$(includedir)/freetype2/freetype/config/ftmodule.h
 	$(INSTALL_SCRIPT) -m 755 $(OBJ_BUILD)/freetype-config \
