--- configure.orig	Fri Mar  9 22:39:47 2012
+++ configure	Fri Mar  9 22:40:57 2012
@@ -5790,7 +5790,7 @@
 		# tested with SunOS 5.5.1 (solaris 2.5.1) and gcc 2.7.2
 		# tested with SunOS 5.10 (solaris 10) and gcc 3.4.3
 		if test "$DFT_LWR_MODEL" = "shared" ; then
-			LOCAL_LDFLAGS="-R \$(LOCAL_LIBDIR):\${libdir}"
+			LOCAL_LDFLAGS="-R \${libdir}"
 			LOCAL_LDFLAGS2="$LOCAL_LDFLAGS"
 		fi
 		if test "$cf_cv_enable_rpath" = yes ; then
@@ -6957,8 +6957,8 @@
 
 # Check to define _XOPEN_SOURCE "automatically"
 
-cf_XOPEN_SOURCE=500
-cf_POSIX_C_SOURCE=199506L
+cf_XOPEN_SOURCE=600
+cf_POSIX_C_SOURCE=200112L
 cf_xopen_source=
 
 case $host_os in #(vi
@@ -7096,7 +7096,7 @@
 	# setting _XOPEN_SOURCE breaks Lynx on SCO Unix / OpenServer
 	;;
 solaris2.1[0-9]) #(vi
-	cf_xopen_source="-D__EXTENSIONS__ -D_XOPEN_SOURCE=$cf_XOPEN_SOURCE"
+	cf_xopen_source="-D__EXTENSIONS__ -D_XOPEN_SOURCE=600"
 	;;
 solaris2.[1-9]) #(vi
 	cf_xopen_source="-D__EXTENSIONS__"
@@ -7813,7 +7813,7 @@
 echo "$as_me:7813: result: $with_widec" >&5
 echo "${ECHO_T}$with_widec" >&6
 if test "$with_widec" = yes ; then
-	LIB_SUFFIX="w${LIB_SUFFIX}"
+	#LIB_SUFFIX="w${LIB_SUFFIX}"
 	cat >>confdefs.h <<\EOF
 #define USE_WIDEC_SUPPORT 1
 EOF
@@ -7864,7 +7864,7 @@
 echo "${ECHO_T}$cf_result" >&6
 
 if test "$cf_result" = yes ; then
-	CPPFLAGS="$CPPFLAGS -D_XOPEN_SOURCE_EXTENDED"
+	CPPFLAGS="$CPPFLAGS"
 elif test "x" != "x" ; then
 	echo "$as_me:7869: checking checking for compatible value versus " >&5
 echo $ECHO_N "checking checking for compatible value versus ... $ECHO_C" >&6
@@ -7907,7 +7907,7 @@
 echo "${ECHO_T}$cf_result" >&6
 	if test "$cf_result" = no ; then
 		# perhaps we can override it - try...
-		CPPFLAGS="$CPPFLAGS -D_XOPEN_SOURCE_EXTENDED="
+		CPPFLAGS="$CPPFLAGS"
 	fi
 fi
 
