--- ./ntpd/refclock_oncore.c.orig	Tue Apr  5 07:35:33 2011
+++ ./ntpd/refclock_oncore.c	Sun Feb 26 21:30:59 2012
@@ -1210,7 +1210,7 @@
 
 	FILE	*fd;
 	char	*cp, *cc, *ca, line[100], units[2], device[64], **cpp;
-	char	*dirs[] = { "/etc/ntp", "/etc", 0 };
+	char	*dirs[] = { "/etc/inet", "/etc", 0 };
 	int	i, sign, lat_flg, long_flg, ht_flg, mode, mask;
 	double	f1, f2, f3;
 
--- ./include/ntp_config.h.orig	Tue Jun  5 19:16:14 2012
+++ ./include/ntp_config.h	Tue Jun  5 19:16:22 2012
@@ -10,7 +10,7 @@
  */
 #ifndef CONFIG_FILE
 # ifndef SYS_WINNT
-#  define	CONFIG_FILE "/etc/ntp.conf"
+#  define	CONFIG_FILE "/etc/inet/ntp.conf"
 # else /* SYS_WINNT */
 #  define	CONFIG_FILE	"%windir%\\system32\\drivers\\etc\\ntp.conf"
 #  define	ALT_CONFIG_FILE "%windir%\\ntp.conf"
