diff -ru ntp-4.2.8p1.orig/include/ntp_config.h ntp-4.2.8p1/include/ntp_config.h
--- include/ntp_config.h.orig	Fri Jan  2 22:47:47 2015
+++ include/ntp_config.h	Mon Mar  2 10:56:10 2015
@@ -14,7 +14,7 @@
  */
 #ifndef CONFIG_FILE
 # ifndef SYS_WINNT
-#  define	CONFIG_FILE "/etc/ntp.conf"
+#  define	CONFIG_FILE "/etc/inet/ntp.conf"
 # else /* SYS_WINNT */
 #  define	CONFIG_FILE	"%windir%\\system32\\drivers\\etc\\ntp.conf"
 #  define	ALT_CONFIG_FILE "%windir%\\ntp.conf"
diff -ru ntp-4.2.8p1.orig/ntpd/refclock_oncore.c ntp-4.2.8p1/ntpd/refclock_oncore.c
--- ntpd/refclock_oncore.c.orig	Fri Jan  2 22:47:48 2015
+++ ntpd/refclock_oncore.c	Mon Mar  2 10:57:03 2015
@@ -1211,7 +1211,7 @@
 
 	FILE	*fd;
 	char	*cc, *ca, line[100], units[2], device[64];
-	const char	*dirs[] = { "/etc/ntp", "/etc", 0 };
+	const char	*dirs[] = { "/etc/inet", "/etc", 0 };
 	const char *cp, **cpp;
 	int	i, sign, lat_flg, long_flg, ht_flg, mode, mask;
 	double	f1, f2, f3;
