--- ntpd/ntpd-opts.c.orig	Wed Feb 22 05:47:38 2012
+++ ntpd/ntpd-opts.c	Sun Feb 26 21:48:46 2012
@@ -366,6 +366,24 @@
 #define NOVIRTUALIPS_FLAGS     (OPTST_DISABLED)
 
 /*
+ *  Mdns option description:
+ */
+#ifdef HAVE_DNSREGISTRATION
+tSCC    zMdnsText[] =
+        "Register with mDNS as a NTP server";
+tSCC    zMdns_NAME[]               = "MDNS";
+tSCC    zMdns_Name[]               = "mdns";
+#define MDNS_FLAGS       (OPTST_DISABLED)
+
+#else   /* disable Mdns */
+#define VALUE_OPT_MDNS NO_EQUIVALENT
+#define MDNS_FLAGS       (OPTST_OMITTED | OPTST_NO_INIT)
+#define zMdnsText       NULL
+#define zMdns_NAME      NULL
+#define zMdns_Name      NULL
+#endif  /* HAVE_DNSREGISTRATION */
+
+/*
  *  modifymmtimer option description:
  */
 #ifdef SYS_WINNT
@@ -837,10 +855,22 @@
      /* desc, NAME, name */ NOVIRTUALIPS_DESC, NOVIRTUALIPS_NAME, NOVIRTUALIPS_name,
      /* disablement strs */ NULL, NULL },
 
-  {  /* entry idx, value */ 15, VALUE_OPT_MODIFYMMTIMER,
-     /* equiv idx, value */ 15, VALUE_OPT_MODIFYMMTIMER,
+  {  /* entry idx, value */ 15, VALUE_OPT_MDNS,
+     /* equiv idx, value */ 15, VALUE_OPT_MDNS,
      /* equivalenced to  */ NO_EQUIVALENT,
      /* min, max, act ct */ 0, 1, 0,
+     /* opt state flags  */ MDNS_FLAGS, 0,
+     /* last opt argumnt */ { NULL },
+     /* arg list/cookie  */ NULL,
+     /* must/cannot opts */ NULL, NULL,
+     /* option proc      */ NULL,
+     /* desc, NAME, name */ zMdnsText, zMdns_NAME, zMdns_Name,
+     /* disablement strs */ NULL, NULL },
+
+  {  /* entry idx, value */ 16, VALUE_OPT_MODIFYMMTIMER,
+     /* equiv idx, value */ 16, VALUE_OPT_MODIFYMMTIMER,
+     /* equivalenced to  */ NO_EQUIVALENT,
+     /* min, max, act ct */ 0, 1, 0,
      /* opt state flags  */ MODIFYMMTIMER_FLAGS, 0,
      /* last opt argumnt */ { NULL }, /* --modifymmtimer */
      /* arg list/cookie  */ NULL,
@@ -849,8 +879,8 @@
      /* desc, NAME, name */ MODIFYMMTIMER_DESC, MODIFYMMTIMER_NAME, MODIFYMMTIMER_name,
      /* disablement strs */ NULL, NULL },
 
-  {  /* entry idx, value */ 16, VALUE_OPT_NOFORK,
-     /* equiv idx, value */ 16, VALUE_OPT_NOFORK,
+  {  /* entry idx, value */ 17, VALUE_OPT_NOFORK,
+     /* equiv idx, value */ 17, VALUE_OPT_NOFORK,
      /* equivalenced to  */ NO_EQUIVALENT,
      /* min, max, act ct */ 0, 1, 0,
      /* opt state flags  */ NOFORK_FLAGS, 0,
@@ -861,8 +891,8 @@
      /* desc, NAME, name */ NOFORK_DESC, NOFORK_NAME, NOFORK_name,
      /* disablement strs */ NULL, NULL },
 
-  {  /* entry idx, value */ 17, VALUE_OPT_NICE,
-     /* equiv idx, value */ 17, VALUE_OPT_NICE,
+  {  /* entry idx, value */ 18, VALUE_OPT_NICE,
+     /* equiv idx, value */ 18, VALUE_OPT_NICE,
      /* equivalenced to  */ NO_EQUIVALENT,
      /* min, max, act ct */ 0, 1, 0,
      /* opt state flags  */ NICE_FLAGS, 0,
@@ -873,8 +903,8 @@
      /* desc, NAME, name */ NICE_DESC, NICE_NAME, NICE_name,
      /* disablement strs */ NULL, NULL },
 
-  {  /* entry idx, value */ 18, VALUE_OPT_PIDFILE,
-     /* equiv idx, value */ 18, VALUE_OPT_PIDFILE,
+  {  /* entry idx, value */ 19, VALUE_OPT_PIDFILE,
+     /* equiv idx, value */ 19, VALUE_OPT_PIDFILE,
      /* equivalenced to  */ NO_EQUIVALENT,
      /* min, max, act ct */ 0, 1, 0,
      /* opt state flags  */ PIDFILE_FLAGS, 0,
@@ -885,8 +915,8 @@
      /* desc, NAME, name */ PIDFILE_DESC, PIDFILE_NAME, PIDFILE_name,
      /* disablement strs */ NULL, NULL },
 
-  {  /* entry idx, value */ 19, VALUE_OPT_PRIORITY,
-     /* equiv idx, value */ 19, VALUE_OPT_PRIORITY,
+  {  /* entry idx, value */ 20, VALUE_OPT_PRIORITY,
+     /* equiv idx, value */ 20, VALUE_OPT_PRIORITY,
      /* equivalenced to  */ NO_EQUIVALENT,
      /* min, max, act ct */ 0, 1, 0,
      /* opt state flags  */ PRIORITY_FLAGS, 0,
@@ -897,8 +927,8 @@
      /* desc, NAME, name */ PRIORITY_DESC, PRIORITY_NAME, PRIORITY_name,
      /* disablement strs */ NULL, NULL },
 
-  {  /* entry idx, value */ 20, VALUE_OPT_QUIT,
-     /* equiv idx, value */ 20, VALUE_OPT_QUIT,
+  {  /* entry idx, value */ 21, VALUE_OPT_QUIT,
+     /* equiv idx, value */ 21, VALUE_OPT_QUIT,
      /* equivalenced to  */ NO_EQUIVALENT,
      /* min, max, act ct */ 0, 1, 0,
      /* opt state flags  */ QUIT_FLAGS, 0,
@@ -909,8 +939,8 @@
      /* desc, NAME, name */ QUIT_DESC, QUIT_NAME, QUIT_name,
      /* disablement strs */ NULL, NULL },
 
-  {  /* entry idx, value */ 21, VALUE_OPT_PROPAGATIONDELAY,
-     /* equiv idx, value */ 21, VALUE_OPT_PROPAGATIONDELAY,
+  {  /* entry idx, value */ 22, VALUE_OPT_PROPAGATIONDELAY,
+     /* equiv idx, value */ 22, VALUE_OPT_PROPAGATIONDELAY,
      /* equivalenced to  */ NO_EQUIVALENT,
      /* min, max, act ct */ 0, 1, 0,
      /* opt state flags  */ PROPAGATIONDELAY_FLAGS, 0,
--- ntpd/ntpd-opts.h.orig	Wed Feb 22 05:47:38 2012
+++ ntpd/ntpd-opts.h	Sun Feb 26 21:45:00 2012
@@ -199,6 +200,10 @@
 #  warning undefining NOVIRTUALIPS due to option name conflict
 #  undef   NOVIRTUALIPS
 # endif
+# ifdef    MDNS
+#  warning undefining MDNS due to option name conflict
+#  undef   MDNS
+# endif
 # ifdef    MODIFYMMTIMER
 #  warning undefining MODIFYMMTIMER due to option name conflict
 #  undef   MODIFYMMTIMER
@@ -287,6 +292,7 @@
 # undef KEYFILE
 # undef LOGFILE
 # undef NOVIRTUALIPS
+# undef MDNS
 # undef MODIFYMMTIMER
 # undef NOFORK
 # undef NICE
@@ -326,6 +332,9 @@
 #define VALUE_OPT_KEYFILE        'k'
 #define VALUE_OPT_LOGFILE        'l'
 #define VALUE_OPT_NOVIRTUALIPS   'L'
+#ifdef HAVE_DNSREGISTRATION
+#define VALUE_OPT_MDNS           'm'
+#endif /* HAVE_DNSREGISTRATION */
 #define VALUE_OPT_MODIFYMMTIMER  'M'
 #define VALUE_OPT_NOFORK         'n'
 #define VALUE_OPT_NICE           'N'
