--- libpcap-1.3.0/configure.in	Thu Mar 29 13:42:04 2012
+++ libpcap-1.3.0.works/configure.in	Thu Jan  3 14:32:29 2013
@@ -356,6 +356,7 @@
 	# Also, due to the bug above applications that link to libpcap with
 	# libdlpi will have to add "-L/lib" option to "configure".
 	#
+	AC_CHECK_HEADERS(sys/bufmod.h sys/dlpi_ext.h)
 	saved_ldflags=$LDFLAGS
 	LDFLAGS="$LIBS -L/lib"
 	AC_CHECK_LIB(dlpi, dlpi_walk,
@@ -573,7 +574,6 @@
 		case "$V_PCAP" in
 
 		dlpi|libdlpi)
-			AC_CHECK_HEADERS(sys/bufmod.h sys/dlpi_ext.h)
 			#
 			# This might be Solaris 8 or later, with
 			# SIOCGLIFCONF, or it might be some other OS
