--- pciutils-3.1.9/Makefile.orig	Sat Jan 14 14:07:55 2012
+++ pciutils-3.1.9/Makefile	Thu May 24 22:24:41 2012
@@ -1,11 +1,12 @@
 # Makefile for The PCI Utilities
 # (c) 1998--2012 Martin Mares <mj@ucw.cz>
 
-OPT=-O2
+OPT=-O2 -DBYTE_ORDER=1234 -DLITTLE_ENDIAN=1234
 CFLAGS=$(OPT) -Wall -W -Wno-parentheses -Wstrict-prototypes -Wmissing-prototypes
 
 VERSION=3.1.9
 DATE=2012-01-14
+SED=gsed
 
 # Host OS and release (override if you are cross-compiling)
 HOST=
@@ -25,7 +26,7 @@
 ABI_VERSION=.3
 
 # Installation directories
-PREFIX=/usr/local
+PREFIX=/usr
 SBINDIR=$(PREFIX)/sbin
 SHAREDIR=$(PREFIX)/share
 IDSDIR=$(SHAREDIR)
@@ -35,8 +36,8 @@
 PKGCFDIR=$(LIBDIR)/pkgconfig
 
 # Commands
-INSTALL=install
-DIRINSTALL=install -d
+INSTALL=/usr/ucb/install
+DIRINSTALL=/usr/ucb/install -d
 STRIP=-s
 CC=$(CROSS_COMPILE)gcc
 AR=$(CROSS_COMPILE)ar
@@ -78,7 +79,7 @@
 common.o: common.c pciutils.h $(PCIINC)
 
 update-pciids: update-pciids.sh
-	sed <$< >$@ "s@^DEST=.*@DEST=$(IDSDIR)/$(PCI_IDS)@;s@^PCI_COMPRESSED_IDS=.*@PCI_COMPRESSED_IDS=$(PCI_COMPRESSED_IDS)@"
+	$(SED) <$< >$@ "s@^DEST=.*@DEST=$(IDSDIR)/$(PCI_IDS)@;s@^PCI_COMPRESSED_IDS=.*@PCI_COMPRESSED_IDS=$(PCI_COMPRESSED_IDS)@"
 	chmod +x $@
 
 # The example of use of libpci
@@ -89,7 +90,7 @@
 	$(CC) $(LDFLAGS) $(TARGET_ARCH) $^ $(LDLIBS) -o $@
 
 %.8 %.7: %.man
-	M=`echo $(DATE) | sed 's/-01-/-January-/;s/-02-/-February-/;s/-03-/-March-/;s/-04-/-April-/;s/-05-/-May-/;s/-06-/-June-/;s/-07-/-July-/;s/-08-/-August-/;s/-09-/-September-/;s/-10-/-October-/;s/-11-/-November-/;s/-12-/-December-/;s/\(.*\)-\(.*\)-\(.*\)/\3 \2 \1/'` ; sed <$< >$@ "s/@TODAY@/$$M/;s/@VERSION@/pciutils-$(VERSION)/;s#@IDSDIR@#$(IDSDIR)#"
+	M=`echo $(DATE) | $(SED) 's/-01-/-January-/;s/-02-/-February-/;s/-03-/-March-/;s/-04-/-April-/;s/-05-/-May-/;s/-06-/-June-/;s/-07-/-July-/;s/-08-/-August-/;s/-09-/-September-/;s/-10-/-October-/;s/-11-/-November-/;s/-12-/-December-/;s/\(.*\)-\(.*\)-\(.*\)/\3 \2 \1/'` ; $(SED) <$< >$@ "s/@TODAY@/$$M/;s/@VERSION@/pciutils-$(VERSION)/;s#@IDSDIR@#$(IDSDIR)#"
 
 clean:
 	rm -f `find . -name "*~" -o -name "*.[oa]" -o -name "\#*\#" -o -name TAGS -o -name core -o -name "*.orig"`
@@ -100,7 +101,10 @@
 
 install: all
 # -c is ignored on Linux, but required on FreeBSD
-	$(DIRINSTALL) -m 755 $(DESTDIR)$(SBINDIR) $(DESTDIR)$(IDSDIR) $(DESTDIR)$(MANDIR)/man8 $(DESTDIR)$(MANDIR)/man7
+	$(DIRINSTALL) -m 755 $(DESTDIR)$(SBINDIR)
+	$(DIRINSTALL) -m 755 $(DESTDIR)$(IDSDIR)
+	$(DIRINSTALL) -m 755 $(DESTDIR)$(MANDIR)/man8
+	$(DIRINSTALL) -m 755 $(DESTDIR)$(MANDIR)/man7
 	$(INSTALL) -c -m 755 $(STRIP) lspci setpci $(DESTDIR)$(SBINDIR)
 	$(INSTALL) -c -m 755 update-pciids $(DESTDIR)$(SBINDIR)
 	$(INSTALL) -c -m 644 $(PCI_IDS) $(DESTDIR)$(IDSDIR)
@@ -113,7 +117,9 @@
 endif
 
 install-lib: $(PCIINC_INS) lib/$(PCILIB) lib/$(PCILIBPC)
-	$(DIRINSTALL) -m 755 $(DESTDIR)$(INCDIR)/pci $(DESTDIR)$(LIBDIR) $(DESTDIR)$(PKGCFDIR)
+	$(DIRINSTALL) -m 755 $(DESTDIR)$(INCDIR)/pci
+	$(DIRINSTALL) -m 755 $(DESTDIR)$(LIBDIR)
+	$(DIRINSTALL) -m 755 $(DESTDIR)$(PKGCFDIR)
 	$(INSTALL) -c -m 644 $(PCIINC_INS) $(DESTDIR)$(INCDIR)/pci
 	$(INSTALL) -c -m 644 lib/$(PCILIB) $(DESTDIR)$(LIBDIR)
 	$(INSTALL) -c -m 644 lib/$(PCILIBPC) $(DESTDIR)$(PKGCFDIR)
