diff -ru expat-2.1.0-orig/configure expat-2.1.0/configure
--- expat-2.1.0-orig/configure	Sat Mar 24 15:12:48 2012
+++ expat-2.1.0/configure	Tue Aug 12 15:10:47 2014
@@ -15161,9 +15161,9 @@
 
 if test "$GCC" = yes ; then
                         OLDCFLAGS="$CFLAGS -Wall -Wmissing-prototypes -Wstrict-prototypes"
-    CFLAGS="$OLDCFLAGS -fexceptions"
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether $CC accepts -fexceptions" >&5
-$as_echo_n "checking whether $CC accepts -fexceptions... " >&6; }
+    CFLAGS="$OLDCFLAGS -funwind-tables"
+    { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether $CC accepts -funwind-tables" >&5
+$as_echo_n "checking whether $CC accepts -funwind-tables... " >&6; }
     cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
diff -ru expat-2.1.0-orig/configure.in expat-2.1.0/configure.in
--- expat-2.1.0-orig/configure.in	Sat Mar  3 18:45:53 2012
+++ expat-2.1.0/configure.in	Tue Aug 12 15:11:41 2014
@@ -67,13 +67,13 @@
 
 if test "$GCC" = yes ; then
     dnl
-    dnl Be careful about adding the -fexceptions option; some versions of
+    dnl Be careful about adding the -funwind-tables option; some versions of
     dnl GCC don't support it and it causes extra warnings that are only
     dnl distracting; avoid.
     dnl
     OLDCFLAGS="$CFLAGS -Wall -Wmissing-prototypes -Wstrict-prototypes"
-    CFLAGS="$OLDCFLAGS -fexceptions"
-    AC_MSG_CHECKING(whether $CC accepts -fexceptions)
+    CFLAGS="$OLDCFLAGS -funwind-tables"
+    AC_MSG_CHECKING(whether $CC accepts -funwind-tables)
     AC_TRY_LINK( , ,
                    AC_MSG_RESULT(yes),
                    AC_MSG_RESULT(no); CFLAGS="$OLDCFLAGS")
