diff -ur pristine.libtool-2.4.2/Makefile.in libtool-2.4.2/Makefile.in
--- pristine.libtool-2.4.2/Makefile.in	Fri Nov 22 21:40:43 2013
+++ libtool-2.4.2/Makefile.in	Fri Nov 22 21:56:03 2013
@@ -2029,7 +2029,7 @@
 	$(MAKE) $(AM_MAKEFLAGS) check-TESTS check-local
 check: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) check-recursive
-all-am: Makefile $(INFO_DEPS) $(LTLIBRARIES) $(SCRIPTS) $(MANS) \
+all-am: Makefile $(LTLIBRARIES) $(SCRIPTS) \
 		$(HEADERS) config.h all-local
 installdirs: installdirs-recursive
 installdirs-am:
@@ -2104,7 +2104,7 @@
 info-am: $(INFO_DEPS)
 
 install-data-am: install-data-local install-includeHEADERS \
-	install-info-am install-ltdlincludeHEADERS install-man
+	 install-ltdlincludeHEADERS
 
 install-dvi: install-dvi-recursive
 
