--- libgcrypt-1.5.0/config.guess.orig	Wed Jun 29 08:52:30 2011
+++ libgcrypt-1.5.0/config.guess	Fri Mar  2 22:12:08 2012
@@ -348,7 +348,7 @@
 	# This test works for both compilers.
 	if [ "$CC_FOR_BUILD" != 'no_compiler_found' ]; then
 	    if (echo '#ifdef __amd64'; echo IS_64BIT_ARCH; echo '#endif') | \
-		(CCOPTS= $CC_FOR_BUILD -E - 2>/dev/null) | \
+		(CCOPTS= $CC_FOR_BUILD $CFLAGS -E - 2>/dev/null) | \
 		grep IS_64BIT_ARCH >/dev/null
 	    then
 		SUN_ARCH="x86_64"
