--- gnutls-3.0.15/build-aux/config.guess.orig	Thu Mar  1 22:44:28 2012
+++ gnutls-3.0.15/build-aux/config.guess	Mon Mar  5 21:46:08 2012
@@ -346,7 +346,7 @@
 	# This test works for both compilers.
 	if [ "$CC_FOR_BUILD" != 'no_compiler_found' ]; then
 	    if (echo '#ifdef __amd64'; echo IS_64BIT_ARCH; echo '#endif') | \
-		(CCOPTS= $CC_FOR_BUILD -E - 2>/dev/null) | \
+		(CCOPTS= $CC_FOR_BUILD $CFLAGS -E - 2>/dev/null) | \
 		grep IS_64BIT_ARCH >/dev/null
 	    then
 		SUN_ARCH="x86_64"
