diff -ru protobuf-c-0.15/configure.ac protobuf-c-0.15.fixed/configure.ac
--- protobuf-c-0.15/configure.ac	Thu Mar 10 14:11:47 2011
+++ protobuf-c-0.15.fixed/configure.ac	Fri Jul  8 15:59:15 2011
@@ -14,6 +14,9 @@
 AC_CHECK_HEADERS(io.h)
 AC_CHECK_HEADERS(sys/uio.h)
 AC_CHECK_HEADERS(unistd.h)
+AC_CHECK_LIB(nsl, gethostbyname, , )
+AC_CHECK_LIB(socket, getprotobyname, , )
+AC_CHECK_LIB(resolv, hstrerror, , )
 
 BUILD_PROTOC_C=1
 AC_ARG_ENABLE(protoc, [  --disable-protoc    Suppress build of protoc_c],
diff -ru protobuf-c-0.15/src/google/protobuf-c/protobuf-c-rpc.c protobuf-c-0.15.fixed/src/google/protobuf-c/protobuf-c-rpc.c
--- protobuf-c-0.15/src/google/protobuf-c/protobuf-c-rpc.c	Thu Mar 10 14:02:48 2011
+++ protobuf-c-0.15.fixed/src/google/protobuf-c/protobuf-c-rpc.c	Fri Jul  8 15:58:18 2011
@@ -17,6 +17,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <alloca.h>
 #include "protobuf-c-rpc.h"
 #include "protobuf-c-data-buffer.h"
 #include "gsklistmacros.h"
