diff --git a/wscript b/wscript
index a63db7d..8bf97c0 100644
--- a/wscript
+++ b/wscript
@@ -378,6 +378,9 @@ def configure(conf):
   conf.check(lib='util', libpath=['/usr/lib', '/usr/local/lib'],
              uselib_store='UTIL')
 
+  conf.check(lib='z', libpath=['/usr/lib', '/usr/local/lib'],
+             uselib_store='ZLIB')
+
   # normalize DEST_CPU from --dest-cpu, DEST_CPU or built-in value
   if Options.options.dest_cpu and Options.options.dest_cpu:
     conf.env['DEST_CPU'] = canonical_cpu_type(Options.options.dest_cpu)
