--- illumos-kvm-cmd-1.0.2/build.sh.orig	Thu Mar 22 19:39:15 2012
+++ illumos-kvm-cmd-1.0.2/build.sh	Thu Mar 22 19:40:01 2012
@@ -10,6 +10,7 @@
 PNGDIR="${PWD}/libpng-1.5.4"
 PNGINC="${PNGDIR}/proto/usr/local/include"
 PNGLIB="${PNGDIR}/proto/usr/local/lib"
+PREFIX="${PREFIX:-/smartdc}"
 
 if [[ ! -d ${PNGDIR} ]]; then
     (curl -k https://download.joyent.com/pub/kvm-cmd/libpng-1.5.4.tar.gz | \
@@ -36,7 +37,9 @@
     --cc=$CC \
     --extra-cflags="-fno-builtin -nodefaultlibs -I${PNGDIR}/proto/usr/local/include" \
     --extra-ldflags="-L${PNGDIR}/proto/usr/local/lib -lz -lm -lc" \
-    --prefix=/smartdc \
+    --prefix=$PREFIX \
+    --sysconfdir=/etc \
+    --install=/usr/gnu/bin/install \
     --audio-card-list= \
     --audio-drv-list= \
     --disable-bluez \
