diff -ur pristine/nspr/configure nss-3.16.1/nspr/configure
--- pristine/nspr/configure	Thu May  1 18:27:29 2014
+++ nss-3.16.1/nspr/configure	Thu May 22 20:57:50 2014
@@ -7627,7 +7627,7 @@
     # be linked against -lrt (or -lposix4) so add it to OS_LIBS
     _rev=`uname -r`
     _librt=`echo $_rev 5.6 | awk '{ if ($1 > $2) print "-lrt"; else print "-lposix4" }'`
-    OS_LIBS="$OS_LIBS $_librt"
+    # OS_LIBS="$OS_LIBS $_librt"
     ;;
 
 *-sco-sysv5*)
--- nss-3.25/nss/coreconf/SunOS5.mk.orig	Wed Jul 13 15:14:12 2016
+++ nss-3.25/nss/coreconf/SunOS5.mk	Wed Jul 13 15:15:34 2016
@@ -34,7 +34,7 @@
 
 ifdef NS_USE_GCC
 	CC         = gcc
-	OS_CFLAGS += -Wall -Wno-format -Werror-implicit-function-declaration -Wno-switch
+	OS_CFLAGS += --std=gnu99 -Wall -Wno-format -Werror-implicit-function-declaration -Wno-switch
 	CCC        = g++
 	CCC       += -Wall -Wno-format
 	ASFLAGS	  += -x assembler-with-cpp
@@ -141,4 +141,4 @@
 RPATH = -R '$$ORIGIN'
 endif
 
-OS_LIBS += -lthread -lnsl -lsocket -lposix4 -ldl -lc
+OS_LIBS += -lnsl -lsocket -ldl -lc
