--- pbzip2-1.1.8/Makefile.orig	Sun Jun 10 13:36:57 2012
+++ pbzip2-1.1.8/Makefile	Mon Dec  8 15:31:58 2014
@@ -37,12 +37,9 @@
 CFLAGS += -pthread
 
 # External libraries
-LDFLAGS = -lbz2
+LDFLAGS += -lbz2
 LDFLAGS += -lpthread
 
-# Where you want pbzip2 installed when you do 'make install'
-PREFIX = /usr
-
 all: pbzip2
 
 # Standard pbzip2 compile
@@ -55,15 +52,15 @@
 
 # Install the binary pbzip2 program and man page
 install: pbzip2
-	if ( test ! -d $(PREFIX)/bin ) ; then mkdir -p $(PREFIX)/bin ; fi
-	if ( test ! -d $(PREFIX)/man ) ; then mkdir -p $(PREFIX)/man ; fi
-	if ( test ! -d $(PREFIX)/man/man1 ) ; then mkdir -p $(PREFIX)/man/man1 ; fi
-	cp -f pbzip2 $(PREFIX)/bin/pbzip2
-	chmod a+x $(PREFIX)/bin/pbzip2
-	ln -s -f $(PREFIX)/bin/pbzip2 $(PREFIX)/bin/pbunzip2
-	ln -s -f $(PREFIX)/bin/pbzip2 $(PREFIX)/bin/pbzcat
-	cp -f pbzip2.1 $(PREFIX)/man/man1
-	chmod a+r $(PREFIX)/man/man1/pbzip2.1
+	if ( test ! -d $(DESTDIR)/$(PREFIX)/bin ) ; then mkdir -p $(DESTDIR)/$(PREFIX)/bin ; fi
+	if ( test ! -d $(DESTDIR)/$(PREFIX)/man ) ; then mkdir -p $(DESTDIR)/$(PREFIX)/man ; fi
+	if ( test ! -d $(DESTDIR)/$(PREFIX)/man/man1 ) ; then mkdir -p $(DESTDIR)/$(PREFIX)/man/man1 ; fi
+	cp -f pbzip2 $(DESTDIR)/$(PREFIX)/bin/pbzip2
+	chmod a+x $(DESTDIR)/$(PREFIX)/bin/pbzip2
+	ln -s -f pbzip2 $(DESTDIR)/$(PREFIX)/bin/pbunzip2
+	ln -s -f pbzip2 $(DESTDIR)/$(PREFIX)/bin/pbzcat
+	cp -f pbzip2.1 $(DESTDIR)/$(PREFIX)/man/man1
+	chmod a+r $(DESTDIR)/$(PREFIX)/man/man1/pbzip2.1
 
 clean:
 	rm -f *.o pbzip2
