diff --git a/erts/emulator/Makefile.in b/erts/emulator/Makefile.in
--- a/erts/emulator/Makefile.in	Wed Apr  9 21:29:24 2014
+++ b/erts/emulator/Makefile.in	Wed Apr  9 21:30:29 2014
@@ -51,7 +51,7 @@
 
 LDFLAGS=@LDFLAGS@
 ARFLAGS=rc
-OMIT_OMIT_FP=no
+OMIT_OMIT_FP=yes
 
 ifeq ($(TYPE),debug)
 PURIFY =
diff --git a/erts/lib_src/Makefile.in b/erts/lib_src/Makefile.in
index 757b3b2..a456f64 100644
--- a/erts/lib_src/Makefile.in
+++ b/erts/lib_src/Makefile.in
@@ -20,7 +20,7 @@
 include $(ERL_TOP)/make/target.mk
 include ../include/internal/$(TARGET)/ethread.mk
 
-OMIT_OMIT_FP=no
+OMIT_OMIT_FP=yes
 
 
 CC=@CC@
