diff --git a/erts/emulator/Makefile.in b/erts/emulator/Makefile.in
index 55b1f41..8e8a027 100644
--- a/erts/emulator/Makefile.in
+++ b/erts/emulator/Makefile.in
@@ -209,7 +209,7 @@ LIB_PREFIX=lib
 LIB_SUFFIX=.a
 endif
 
-OMIT_OMIT_FP=no
+OMIT_OMIT_FP=yes
 
 ifeq (@EMU_LOCK_CHECKING@,yes)
 NO_INLINE_FUNCTIONS=true
diff --git a/erts/lib_src/Makefile.in b/erts/lib_src/Makefile.in
index 757b3b2..a456f64 100644
--- a/erts/lib_src/Makefile.in
+++ b/erts/lib_src/Makefile.in
@@ -20,7 +20,7 @@
 include $(ERL_TOP)/make/target.mk
 include ../include/internal/$(TARGET)/ethread.mk
 
-OMIT_OMIT_FP=no
+OMIT_OMIT_FP=yes
 
 
 CC=@CC@
