diff -ru libffi-3.1-orig/configure libffi-3.1/configure
--- libffi-3.1-orig/configure	Mon May 19 09:44:03 2014
+++ libffi-3.1/configure	Tue Aug 12 20:18:42 2014
@@ -16944,7 +16944,7 @@
 fi
 
 if test "x$GCC" = "xyes"; then
-  CFLAGS="$CFLAGS -fexceptions"
+  CFLAGS="$CFLAGS -funwind-tables"
 fi
 
 cat > local.exp <<EOF
@@ -18661,7 +18661,7 @@
 
   	libffi_cv_ro_eh_frame=no
   	echo 'extern void foo (void); void bar (void) { foo (); foo (); }' > conftest.c
-  	if $CC $CFLAGS -c -fpic -fexceptions -o conftest.o conftest.c > /dev/null 2>&1; then
+  	if $CC $CFLAGS -c -fpic -funwind-tables -o conftest.o conftest.c > /dev/null 2>&1; then
 	    objdump -h conftest.o > conftest.dump 2>&1
 	    libffi_eh_frame_line=`grep -n eh_frame conftest.dump | cut -d: -f 1`
 	    libffi_test_line=`expr $libffi_eh_frame_line + 1`p
