diff -ruN Python-2.6.7-orig/configure.in Python-2.6.7/configure.in
--- Python-2.6.7-orig/configure.in	2010-05-25 02:27:03.000000000 +0000
+++ Python-2.6.7/configure.in	2012-02-15 21:28:00.409780555 +0000
@@ -372,7 +372,7 @@
   # except for Solaris 10, where it must not be defined, 
   # as it implies XPG4.2
   case $ac_sys_system/$ac_sys_release in
-    SunOS/5.10)
+    SunOS/5.10|SunOS/5.11)
       ;;
     *)
       AC_DEFINE(_XOPEN_SOURCE_EXTENDED, 1,
