--- Python-2.6.7/configure.in	Wed Feb 22 18:07:20 2012
+++ Python-2.6.7/configure.in	Wed Feb 22 18:07:27 2012
@@ -1931,7 +1931,11 @@
 
 
 # checks for libraries
-AC_CHECK_LIB(dl, dlopen)	# Dynamic linking for SunOS/Solaris and SYSV
+case  $ac_sys_system/$ac_sys_release in
+  SunOS/5.11) ;;
+  *) AC_CHECK_LIB(dl, dlopen)	# Dynamic linking for SunOS/Solaris and SYSV
+  ;;
+esac
 AC_CHECK_LIB(dld, shl_load)	# Dynamic linking for HP-UX
 
 # only check for sem_init if thread support is requested
