--- sudo-1.8.7/src/Makefile.in.orig	2013-06-04 09:49:25.000000000 +0000
+++ sudo-1.8.7/src/Makefile.in	2013-09-27 15:41:26.700037447 +0000
@@ -117,7 +117,7 @@
 	$(LIBTOOL) --mode=link $(CC) -o $@ $(OBJS) $(LDFLAGS) $(PIE_LDFLAGS) $(SSP_LDFLAGS) $(LIBS)
 
 libsudo_noexec.la: sudo_noexec.lo
-	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(LT_LDFLAGS) -o $@ sudo_noexec.lo -avoid-version -rpath $(noexecdir)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(LT_LDFLAGS) -o $@ sudo_noexec.lo -avoid-version @LIBS@ -rpath $(noexecdir)
 
 sesh: $(SESH_OBJS) @LIBINTL@ $(LT_LIBS)
 	$(LIBTOOL) --mode=link $(CC) -o $@ $(SESH_OBJS) $(LDFLAGS) $(PIE_LDFLAGS) $(SSP_LDFLAGS) @LIBINTL@ $(LIBS)
--- sudo-1.8.7/plugins/group_file/Makefile.in.orig	2013-06-04 09:49:25.000000000 +0000
+++ sudo-1.8.7/plugins/group_file/Makefile.in	2013-09-27 15:42:13.330188598 +0000
@@ -103,7 +103,7 @@
 	@sed 's/^/+e /' $(shlib_exp) > $@
 
 group_file.la: $(OBJS) $(LT_LIBS) @LT_LDDEP@
-	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(LT_LDFLAGS) -o $@ $(OBJS) $(LIBS) -module -avoid-version -rpath $(plugindir)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(LT_LDFLAGS) -o $@ $(OBJS) $(LIBS) -module -avoid-version @LIBS@ -rpath $(plugindir)
 
 pre-install:
 
--- sudo-1.8.7/plugins/sample/Makefile.in.orig	2013-06-04 09:48:28.000000000 +0000
+++ sudo-1.8.7/plugins/sample/Makefile.in	2013-09-27 15:41:26.700325769 +0000
@@ -34,7 +34,7 @@
 INSTALL = $(SHELL) $(top_srcdir)/install-sh -c
 
 # Libraries
-LIBS = $(LIBOBJDIR)/libreplace.la
+LIBS = $(LIBOBJDIR)/libreplace.la @LIBS@
 
 # C preprocessor flags
 CPPFLAGS = -I$(incdir) -I$(top_builddir) -I$(top_srcdir) @CPPFLAGS@
--- sudo-1.8.7/plugins/sudoers/Makefile.in.orig	2013-06-04 09:49:25.000000000 +0000
+++ sudo-1.8.7/plugins/sudoers/Makefile.in	2013-09-27 15:41:26.701009790 +0000
@@ -47,7 +47,7 @@
 
 # Libraries
 LT_LIBS = $(top_builddir)/common/libcommon.la $(LIBOBJDIR)libreplace.la
-LIBS = $(LT_LIBS) @LIBINTL@
+LIBS = $(LT_LIBS) @LIBINTL@ @LIBS@
 NET_LIBS = @NET_LIBS@
 SUDOERS_LIBS = @SUDOERS_LIBS@ @AFS_LIBS@ @GETGROUPS_LIB@ $(LIBS) $(NET_LIBS) @ZLIB@
 REPLAY_LIBS = @REPLAY_LIBS@ @ZLIB@
--- sudo-1.8.7/plugins/system_group/Makefile.in.orig	2013-06-04 09:48:31.000000000 +0000
+++ sudo-1.8.7/plugins/system_group/Makefile.in	2013-09-27 15:41:26.701271848 +0000
@@ -103,7 +103,7 @@
 	@sed 's/^/+e /' $(shlib_exp) > $@
 
 system_group.la: $(OBJS) $(LT_LIBS) @LT_LDDEP@
-	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(LT_LDFLAGS) -o $@ $(OBJS) $(LIBS) -module -avoid-version -rpath $(plugindir)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(LT_LDFLAGS) -o $@ $(OBJS) $(LIBS) -module -avoid-version @LIBS@ -rpath $(plugindir)
 
 pre-install:
 
