--- p7zip_9.20.1/makefile.orig	Sun Mar 13 11:52:45 2011
+++ p7zip_9.20.1/makefile	Mon Feb 27 17:06:25 2012
@@ -1,11 +1,5 @@
 MKDIR = mkdir -p
 
-DEST_HOME=/usr/local
-DEST_BIN=$(DEST_HOME)/bin
-DEST_SHARE=$(DEST_HOME)/lib/p7zip
-DEST_SHARE_DOC=$(DEST_HOME)/share/doc/p7zip
-DEST_MAN=$(DEST_HOME)/man
-
 .PHONY: default all all2 7za 7zG 7zFM sfx 7z 7zr Client7z common common7z clean tar_bin depend test test_7z test_7zr test_7zG test_Client7z all_test app
 
 default:7za
