--- p7zip_16.02-orig/makefile.common	Thu May 19 06:20:02 2016
+++ p7zip_16.02/makefile.common	Wed Oct 12 12:35:02 2016
@@ -1,10 +1,10 @@
 MKDIR = mkdir -p
 
-DEST_HOME=/usr/local
-DEST_BIN=$(DEST_HOME)/bin
-DEST_SHARE=$(DEST_HOME)/lib/p7zip
-DEST_SHARE_DOC=$(DEST_HOME)/share/doc/p7zip
-DEST_MAN=$(DEST_HOME)/man
+#DEST_HOME=/usr/local
+#DEST_BIN=$(DEST_HOME)/bin
+#DEST_SHARE=$(DEST_HOME)/lib/p7zip
+#DEST_SHARE_DOC=$(DEST_HOME)/share/doc/p7zip
+#DEST_MAN=$(DEST_HOME)/man
 
 .PHONY: default all all2 7za 7zG 7zFM sfx 7z 7zr Client7z common common7z LangAndHelp tar_bin depend test test_7z test_7zr test_7zG test_Client7z LzmaCon test_LzmaCon  all_test app cppcheck lzham test_lib other
 
