diff -urN postgresql-8.4.0.orig/contrib/citext/citext.sql.in postgresql-8.4.0/contrib/citext/citext.sql.in
--- postgresql-8.4.0.orig/contrib/citext/citext.sql.in	2008-09-05 20:25:16.000000000 +0200
+++ postgresql-8.4.0/contrib/citext/citext.sql.in	2009-08-13 14:09:49.000000000 +0200
@@ -1,7 +1,9 @@
 /* $PostgreSQL: pgsql/contrib/citext/citext.sql.in,v 1.3 2008/09/05 18:25:16 tgl Exp $ */
 
 -- Adjust this setting to control where the objects get created.
-SET search_path = public;
+BEGIN;
+CREATE SCHEMA citext;
+SET search_path = citext, public;
 
 --
 --  PostgreSQL code for CITEXT.
@@ -486,3 +488,5 @@
 CREATE OR REPLACE FUNCTION translate( citext, citext, text ) RETURNS TEXT AS $$
     SELECT pg_catalog.translate( pg_catalog.translate( $1::pg_catalog.text, pg_catalog.lower($2::pg_catalog.text), $3), pg_catalog.upper($2::pg_catalog.text), $3);
 $$ LANGUAGE SQL IMMUTABLE STRICT;
+
+COMMIT;
