Index: Makefile
===================================================================
--- Makefile.orig	Mon Aug 29 21:19:27 2011
+++ Makefile	Fri Feb 24 05:57:56 2012
@@ -169,6 +169,8 @@
 	echo Installing cups-config script...
 	$(INSTALL_DIR) -m 755 $(BINDIR)
 	$(INSTALL_SCRIPT) cups-config $(BINDIR)/cups-config
+	$(INSTALL_DIR) $(BUILDROOT)$(libdir)/pkgconfig
+	$(INSTALL_DATA) cups.pc $(BUILDROOT)$(libdir)/pkgconfig/cups.pc
 
 
 #

Index: cups.pc.in
===================================================================
--- cups.pc.in	(revision 0)
+++ cups.pc.in	(revision 0)
@@ -0,0 +1,18 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+bindir=@bindir@
+includedir=@includedir@
+libdir=@libdir@
+imagelibdir=@libdir@
+datarootdir=@datadir@
+datadir=@datadir@
+sysconfdir=@sysconfdir@
+cups_datadir=@CUPS_DATADIR@
+cups_serverbin=@CUPS_SERVERBIN@
+cups_serverroot=@CUPS_SERVERROOT@
+
+Name: cups
+Description: Common Unix Printing System
+Version: @CUPS_VERSION@
+Cflags: -I${includedir}
+Libs: -L${libdir} -lcupsimage -lcups

Property changes on: cups.pc.in
___________________________________________________________________
Name: svn:executable
   + *

Index: configure.in
===================================================================
--- configure.in.orig	Wed Aug 31 01:36:33 2011
+++ configure.in	Fri Feb 24 05:55:53 2012
@@ -89,6 +89,7 @@
 	  scheduler/org.cups.cups-lpd.plist
  	  templates/header.tmpl
           packaging/cups.list
+          cups.pc
 	  $LANGFILES)
 
 chmod +x cups-config

Index: configure
===================================================================
--- configure.orig	Fri Feb 24 06:17:02 2012
+++ configure	Fri Feb 24 06:17:30 2012
@@ -22968,7 +22968,7 @@
 
 
 
-ac_config_files="$ac_config_files Makedefs conf/cupsd.conf conf/mime.convs conf/pam.std conf/snmp.conf cups-config data/testprint desktop/cups.desktop doc/help/ref-cupsd-conf.html doc/help/standard.html doc/index.html man/client.conf.man man/cups-deviced.man man/cups-driverd.man man/cups-lpd.man man/cupsaddsmb.man man/cupsd.conf.man man/cupsd.man man/lpoptions.man scheduler/cups-lpd.xinetd scheduler/cups.sh scheduler/cups.xml scheduler/org.cups.cups-lpd.plist templates/header.tmpl packaging/cups.list $LANGFILES"
+ac_config_files="$ac_config_files Makedefs conf/cupsd.conf conf/mime.convs conf/pam.std conf/snmp.conf cups-config data/testprint desktop/cups.desktop doc/help/ref-cupsd-conf.html doc/help/standard.html doc/index.html man/client.conf.man man/cups-deviced.man man/cups-driverd.man man/cups-lpd.man man/cupsaddsmb.man man/cupsd.conf.man man/cupsd.man man/lpoptions.man scheduler/cups-lpd.xinetd scheduler/cups.sh scheduler/cups.xml scheduler/org.cups.cups-lpd.plist templates/header.tmpl packaging/cups.list cups.pc $LANGFILES"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -23548,6 +23548,7 @@
     "scheduler/org.cups.cups-lpd.plist") CONFIG_FILES="$CONFIG_FILES scheduler/org.cups.cups-lpd.plist" ;;
     "templates/header.tmpl") CONFIG_FILES="$CONFIG_FILES templates/header.tmpl" ;;
     "packaging/cups.list") CONFIG_FILES="$CONFIG_FILES packaging/cups.list" ;;
+    "cups.pc") CONFIG_FILES="$CONFIG_FILES cups.pc" ;;
     "$LANGFILES") CONFIG_FILES="$CONFIG_FILES $LANGFILES" ;;
 
   *) { { echo "$as_me:$LINENO: error: invalid argument: $ac_config_target" >&5
